File: java/fib/src/main/java/us/yellosoft/fib/Fib.java
package us.yellosoft.fib;

import java.util.Scanner;
import java.io.IOException;
import java.nio.charset.StandardCharsets;

/** Fibonacci */
public final class Fib {
  /** utility class */
  private Fib() {}

  /**
     <p>Fibonacci</p>
     @param n input
     @return fib(n)
  */
  public static int fib(final int n) {
    if (n < 2) {
      return n;
    } else {
      int a = 0, b = 1;
      int m = n - 1;

      for (int i = 0; i < m; i++) {
        int c = a;
        a = b;
        b = c + b;
      }

      return b;
    }
  }

  /**
     <p>Timed Fibonacci test</p>
     @param args CLI args
     @throws IOException on CLI read error
  */
  public static void main(final String[] args) throws IOException {
    System.out.print("N = ");

    try (Scanner scanner = new Scanner(System.in, StandardCharsets.UTF_8.name())) {
      int n = scanner.nextInt();

      System.out.println("Running");

      int startt = (int) System.currentTimeMillis();

      int x = fib(n);

      int endt = (int) System.currentTimeMillis();

      System.out.println("Fib " + n + " = " + x);
      System.out.println("Time = " + ((endt - startt) / 1000) + " sec");
    }
  }
}

File: system/rust/fib/src/fib.rs
//! Fibonacci

fn fib(n: i64) -> i64 {
    match n {
        0 => 0,
        1 => 1,
        _ => fib(n - 1) + fib(n - 2),
    }
}

#[test]
fn first_five_elements_are_correct() {
    assert_eq!(fib(0), 0);
    assert_eq!(fib(1), 1);
    assert_eq!(fib(2), 1);
    assert_eq!(fib(3), 2);
    assert_eq!(fib(4), 3);
}

fn main() {
    println!("Fib 40 = {}", fib(40));
}
